### PR TITLE
SMB Multichannel

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml [hunk 1]
```diff
@@ -145,6 +146,25 @@
...
+      <AccountKey>[ACCOUNT-KEY]</AccountKey>
+
+      <BlobServiceEndpoint>http://[ACCOUNT].blob.core.windows.net</BlobServiceEndpoint>
+      <QueueServiceEndpoint>http://[ACCOUNT].queue.core.windows.net</QueueServiceEndpoint>
+      <TableServiceEndpoint>http://[ACCOUNT].table.core.windows.net</TableServiceEndpoint>
+      <FileServiceEndpoint>http://[ACCOUNT].file.core.windows.net</FileServiceEndpoint>
+      <BlobServiceSecondaryEndpoint>http://[ACCOUNT]-secondary.blob.core.windows.net</BlobServiceSecondaryEndpoint>
+      <QueueServiceSecondaryEndpoint>http://[ACCOUNT]-secondary.queue.core.windows.net</QueueServiceSecondaryEndpoint>
+      <FileServiceSecondaryEndpoint>http://[ACCOUNT]-secondary.file.core.windows.net</FileServiceSecondaryEndpoint>
+      <TableServiceSecondaryEndpoint>http://[ACCOUNT]-secondary.table.core.windows.net</TableServiceSecondaryEndpoint>
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml:165]: unrelated, but we can think about using <AccountName> to programaticaly compute these endpoints if they're not provided. That would make config less bloated. (We should keep option to override default endpoints though).

File: sdk/storage/Azure.Storage.Files.Shares/swagger/readme.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 ## Configuration
 ``` yaml
 # Generate file storage
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/storage-dataplane-preview/specification/storage/data-plane/Microsoft.FileStorage/preview/2020-02-10/file.json
+input-file: C:\Users\Sean\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.FileStorage\preview\2019-12-12\file.json
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Files.Shares/swagger/readme.md:7]: local path.

File: sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs [hunk 3]
```diff
@@ -720,6 +726,16 @@ public enum ShareTraits
...
+    public partial class SMB
+    {
+        public SMB() { }
+        public Azure.Storage.Files.Shares.Models.SmbMultichannel Multichannel { get { throw null; } set { } }
+    }
+    public partial class SmbMultichannel
+    {
+        public SmbMultichannel() { }
+        public bool? Enabled { get { throw null; } set { } }
+    }
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs:738]: This is quite a lot of wrapping around single boolean. 
For the SMB->SmbMultichannel. How about we do SMBSettings/SmbSettings? { bool MultichannelEnabled; }?
For the ProtocolSettings -> SMB - are we going to have more protocols in the future ?

Btw. these all models seem to be coming from generator. Do we expose them or are we going to write some models by hand?

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml:165]: Maybe in the future.

File: sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs:738]: The xml structure is:
```
    <ProtocolSettings> 
        <SMB> 
            <Multichannel> 
                <Enabled>true|false</Enabled> 
            </Multichannel> 
        </SMB> 
    </ProtocolSettings> 
```

I think we need to leave both the `SMB` and `Multichannel` layers.  If we don't, we'll have a real problem if they add sibling properties in the future.

File: sdk/storage/Azure.Storage.Files.Shares/swagger/readme.md (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/swagger/readme.md]: Fixed!

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**